presto-main/src/main/java/io/prestosql/operator/exchange/LocalExchangeMemoryManager.java

```diff
@@ -22,10 +22,12 @@
 import java.util.concurrent.atomic.AtomicLong;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
 
 @ThreadSafe
 public class LocalExchangeMemoryManager
 {
+    private static final ListenableFuture<?> NOT_BLOCKED = immediateFuture(null);
     private static final SettableFuture<?> NOT_FULL;
```

Comment:
let's remove this. This makes he code unnecessarily complicated.

Reply:
After migrating all checks against the `NOT_FULL` sentinel, it became clear that `null` would be a more clear fit. I've changed it to avoid having `NOT_FULL` and using null to represent no callback registered instead.